Mathlib/Analysis/SpecialFunctions/MulExpNegMulSqIntegral.lean

```diff
@@ -0,0 +1,252 @@
+      BoundedContinuousFunction.coe_add, BoundedContinuousFunction.coe_one, coe_neg,
+      BoundedContinuousFunction.coe_smul, smul_eq_mul, Pi.add_apply, Pi.one_apply, Pi.neg_apply,
+      norm_mul, norm_eq_abs, norm_pow]
+    refine mul_le_of_le_of_le_one' (norm_coe_le_norm g x) ?_
+      (pow_nonneg (abs_nonneg _) n) (norm_nonneg _)
+    apply pow_le_one₀ (abs_nonneg _)
+    rw [mul_assoc, inv_mul_eq_div, abs_le]
+    refine ⟨?_, (add_le_iff_nonpos_right 1).mpr (Left.neg_nonpos_iff.mpr
+      (div_nonneg (mul_nonneg (le_of_lt hε) (mul_self_nonneg (g x))) (le_of_lt hnpos)))⟩
+    apply le_trans (by linarith) (sub_nonneg_of_le ((div_le_one hnpos).mpr _))
+    apply le_trans (le_trans _ (le_of_lt hgN)) (Nat.cast_le.mpr hn)
+    apply mul_le_mul (Preorder.le_refl ε) _ (mul_self_nonneg (g x)) (le_of_lt hε)
+    rw [← abs_le_iff_mul_self_le, abs_norm]
+    exact norm_coe_le_norm g x
+  · apply Eventually.of_forall
+    intro x
+    apply Tendsto.const_mul (g x)
+    simp [mul_assoc, inv_mul_eq_div, ← neg_div]
+    exact tendsto_one_plus_div_pow_exp (-(ε * (g x * g x)))
+
+theorem integral_mulExpNegMulSq_comp_eq {P' : Measure E} [IsFiniteMeasure P']
+    {A : Subalgebra ℝ C(E, ℝ)} (hε : 0 < ε)
+    (hbound : ∀ g ∈ A, ∃ C, ∀ x y : E, dist (g x) (g y) ≤ C)
+    (heq : ∀ g ∈ A, ∫ x, (g : E → ℝ) x ∂P = ∫ x, (g : E → ℝ) x ∂P') {g : C(E, ℝ)} (hgA : g ∈ A) :
+    ∫ x, mulExpNegMulSq ε (g x) ∂P
+        = ∫ x, mulExpNegMulSq ε (g x) ∂P' := by
+  obtain ⟨C, h⟩ := hbound g hgA
+  have one_plus_inv_mul_mem (n : ℕ) : g * (1 + (n : ℝ)⁻¹ • -(ε • g * g)) ^ n ∈ A := by
+    apply Subalgebra.mul_mem A hgA (Subalgebra.pow_mem A _ n)
+    apply Subalgebra.add_mem A (Subalgebra.one_mem A) (Subalgebra.smul_mem A _ n⁻¹)
+    exact Subalgebra.neg_mem A (Subalgebra.mul_mem A (Subalgebra.smul_mem A hgA ε) hgA)
+  have limP : Tendsto (fun n : ℕ => ∫ x, (g * (1 + (n : ℝ)⁻¹ • -(ε • g * g)) ^ n) x ∂P) atTop
+      (𝓝 (∫ x, mulExpNegMulSq ε (g x) ∂P')) := by
+    rw [funext fun n => heq _ (one_plus_inv_mul_mem n)]
+    exact tendsto_integral_one_plus_inv_mul (mkOfBound g C h) hε
+  apply tendsto_nhds_unique (tendsto_integral_one_plus_inv_mul (mkOfBound g C h) hε) limP
+
+theorem abs_integral_sub_setIntegral_mulExpNegMulSq_comp_lt (f : C(E, ℝ))
+    {K : Set E} (hK : MeasurableSet K) (hε : 0 < ε) (hKP : P Kᶜ < ε.toNNReal) :
+    |∫ x, mulExpNegMulSq ε (f x) ∂P
+    - ∫ x in K, mulExpNegMulSq ε (f x) ∂P| < sqrt ε := by
+  apply lt_of_le_of_lt (norm_integral_sub_setIntegral_le
+    (Eventually.of_forall (fun _ => abs_mulExpNegMulSq_le hε)) hK
+    (integrable_mulExpNegMulSq_comp f hε))
+  rw [mul_inv_lt_iff₀ (sqrt_pos_of_pos hε), mul_self_sqrt (le_of_lt hε)]
+  exact toReal_lt_of_lt_ofReal hKP
+
+theorem abs_setIntegral_mulExpNegMulSq_comp_sub_le_mul_measure {K : Set E} (hK : IsCompact K)
+    (hKmeas : MeasurableSet K) (f g : C(E, ℝ)) {δ : ℝ} (hε : 0 < ε)
+    {P : Measure E} [hP : IsFiniteMeasure P] (hfg : ∀ x ∈ K, |g x - f x| < δ) :
```

Comment:
```suggestion
    (hfg : ∀ x ∈ K, |g x - f x| < δ) :
```
There is already a finite measure P in a variable line somewhere above, right? (and I think you don't use the name `hP`)